src/settings/OrderNumberForm.js

```diff
@@ -0,0 +1,170 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import {
+  Field,
+  getFormValues,
+} from 'redux-form';
+
+import { get } from 'lodash';
+
+import stripesForm from '@folio/stripes/form';
+import {
+  Accordion,
+  AccordionSet,
+  Button,
+  Checkbox,
+  Col,
+  Headline,
+  MultiSelection,
+  Pane,
+  Row,
+} from '@folio/stripes/components';
+
+class OrderNumberForm extends Component {
+  static propTypes = {
+    label: PropTypes.object.isRequired,
+    handleSubmit: PropTypes.func.isRequired,
+    pristine: PropTypes.bool.isRequired,
+    submitting: PropTypes.bool.isRequired,
+    stripes: PropTypes.shape({
+      store: PropTypes.object.isRequired,
+    }).isRequired,
+    dispatch: PropTypes.func.isRequired,
+    change: PropTypes.func.isRequired,
+  };
+
+  getLastMenu = () => {
+    const { pristine, submitting } = this.props;
+
+    return (
+      <Button
+        type="submit"
+        buttonStyle="primary"
+        disabled={(pristine || submitting)}
+        marginBottom0
+      >
+        <FormattedMessage id="ui-orders.settings.poNumber.saveBtn" />
+      </Button>
+    );
+  };
+
+  addPrefix = ({ inputValue }) => {
+    const { dispatch, change, stripes: { store } } = this.props;
+    const formValues = getFormValues('orderNumberForm')(store.getState());
+    const prefixList = this.getOptionsList('prefixList');
+    const selectedPrefix = get(formValues, 'prefix', []);
+
+    prefixList.push({ label: inputValue, value: inputValue });
+    selectedPrefix.push({ label: inputValue, value: inputValue });
+    dispatch(change('prefixList', prefixList));
+    dispatch(change('prefix', selectedPrefix));
+  }
+
+  addSuffix = ({ inputValue }) => {
+    const { dispatch, change, stripes: { store } } = this.props;
+    const suffixList = this.getOptionsList('suffixList');
+    const formValues = getFormValues('orderNumberForm')(store.getState());
+    const selectedSuffix = get(formValues, 'prefix', []);
+
+    suffixList.push({ label: inputValue, value: inputValue });
```

Comment:
I think you might need to use `...` here to not modify existing array. Because you're dispatching the new value later.
```
const newOption = {
  label: inputValue,
  value: inputValue,
};
const newSuffixes = [newOption, ...suffixList];`